osidb/models.py

```diff
@@ -1788,6 +1804,98 @@ class Meta:
         )
 
 
+class FlawReferenceManager(ACLMixinManager, TrackingMixinManager):
+    """flawreference manager"""
+
+    @staticmethod
+    def create_flawreference(flaw, url, **extra_fields):
+        """return a new flawreference or update an existing flawreference without saving"""
+        try:
+            flawreference = FlawReference.objects.get(flaw=flaw, url=url)
+            for attr, value in extra_fields.items():
+                setattr(flawreference, attr, value)
+            return flawreference
+
+        except ObjectDoesNotExist:
+            return FlawReference(flaw=flaw, url=url, **extra_fields)
+
+
+class FlawReference(TrackingMixin, ACLMixin, AlertMixin):
+    """Model representing flaw references"""
+
+    class FlawReferenceType(models.TextChoices):
+        """
+        Allowable references:
+
+        ARTICLE:
+            A link to a Security Bulletin or Knowledge Base Article specifically
+            discussing this flaw on the Customer Portal. It always begins with
+            "https://accesss.redhat.com/". It must be a Security Bulletin
+            for Major Incidents. More general articles like hardening should be
+            linked instead in EXTERNAL REFERENCES.
+
+        EXTERNAL_REFERENCES:
+            URL links to other trustworthy sources of information about this
+            vulnerability. A link should not point to a missing resource.
+            Since these links are displayed on the CVE page of the flaw, we only
+            want to include respectable sources (such as upstream advisories,
+            analysis of security researches, etc.).
+        """
+
+        ARTICLE = "ARTICLE"
+        EXTERNAL_REFERENCES = "EXTERNAL_REFERENCES"
```

Comment:
calling the type `EXTERNAL_REFERENCES` seems redundant when the class is already called `FlawReference`

Reply:
Do you prefer to change a constant name or class name? `EXTERNAL` instead of `EXTERNAL_REFERENCES` does not sound good to me (maybe `EXTERNAL_LINKS` to keep it different?).

Reply:
`external`, `generic`, `other` are all fine, but links/references in the name makes no sense as each object represents a single link

Reply:
Oh, you are right; this reasoning makes sense. Thanks for the recommendation!